### DISCUSSION TITLE
[@types/lodash] `ExpChain` should not exclude `undefined` and `null`

### DISCUSSION THREAD

**Discussion by uinz** [DefinitelyTyped/DefinitelyTyped#62031] [Issues with a @types package] [answered]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/common/common.d.ts#L10

<img src="https://user-images.githubusercontent.com/12208108/187421195-031769d0-8275-4248-bc96-52360b5558b8.png" width="400">

<img src="https://user-images.githubusercontent.com/12208108/187421068-c1bfec9a-9570-4b9f-9dc4-001b1fadfe0a.png" width="400">

it's should be `string | undefined`

---

`_.get` is corect

<img src="https://user-images.githubusercontent.com/12208108/187421291-fb60ccd8-5f67-4d1f-a44a-d963c7cf6bf3.png" width="400">

[a comment by an automated account is not shown]

**Comment by aj-r** [comment, marked as answer]
This actually used to work, but the chain types were overhauled a while ago because they were painfully slow. The new types are much faster but don't cover all use cases correctly.

This should be possible to fix but will require significant effort - we would need to create a new `ObjectNullableChain<T>` similar to `ObjectChain<T>`. See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/common/seq.d.ts#L35

Feel free to send a PR.
